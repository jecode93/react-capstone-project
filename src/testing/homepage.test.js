describe('Selectors in weatherSlice', () => {
  const mockState = {
    weather: [
      { main: { temp: 27.44 } },
      { name: 'Haiti' },
    ],
    isLoading: false,
    error: null,
  };

  it('Weather should return the weather data', () => {
    const selectedData = mockState.weather;
    expect(selectedData).toEqual(mockState.weather);
  });

  it('MockState should return the weather status', () => {
    const selectedStatus = mockState.isLoading;
    expect(selectedStatus).toBe(mockState.isLoading);
  });

  it('MockState should return the weather error', () => {
    const selectedError = mockState.error;
    expect(selectedError).toBe(mockState.error);
  });
});
