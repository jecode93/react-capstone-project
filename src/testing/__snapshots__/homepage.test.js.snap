// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Missions component is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="text-2xl"
      >
        <div
          class="second-color relative py-5"
        >
          <p
            class="text-center text-white uppercase"
          >
            Weather view
          </p>
        </div>
        <div
          class="text-white items-center flex justify-around py-20 px-5"
        >
          <img
            alt=""
            src="world.png"
            width="200"
          />
          <h2
            class="text-white text-3xl font-extrabold text-lg tracking-wide text-center my-2 px-4 align-bottom uppercase"
          >
            Weather
            <br />
            World
          </h2>
        </div>
        <hr />
        <h1
          class="text-white text-lg tracking-wide my-2 px-4 align-bottom uppercase"
        >
          Weather by Country
        </h1>
        <div
          class="grid grid-cols-2 md:grid-cols-4"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="text-2xl"
    >
      <div
        class="second-color relative py-5"
      >
        <p
          class="text-center text-white uppercase"
        >
          Weather view
        </p>
      </div>
      <div
        class="text-white items-center flex justify-around py-20 px-5"
      >
        <img
          alt=""
          src="world.png"
          width="200"
        />
        <h2
          class="text-white text-3xl font-extrabold text-lg tracking-wide text-center my-2 px-4 align-bottom uppercase"
        >
          Weather
          <br />
          World
        </h2>
      </div>
      <hr />
      <h1
        class="text-white text-lg tracking-wide my-2 px-4 align-bottom uppercase"
      >
        Weather by Country
      </h1>
      <div
        class="grid grid-cols-2 md:grid-cols-4"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
